cogeo_mosaic/backends/base.py

```diff
@@ -64,3 +69,83 @@ def write(self):
     @abc.abstractmethod
     def update(self):
         """Update existing MosaicJSON on backend."""
+
+    def _update_quadkey(self, quadkey: str, dataset: List[str]):
+        """Update quadkey list."""
+        self.mosaic_def.tiles[quadkey] = dataset
+
+    def _update_metadata(self, bounds: List[float], version: str):
+        """Update bounds and center."""
+        self.mosaic_def.version = version
+        self.mosaic_def.bounds = bounds
+        self.mosaic_def.center = (
+            (bounds[0] + bounds[2]) / 2,
+            (bounds[1] + bounds[3]) / 2,
+            self.mosaic_def.minzoom,
+        )
+
+    def _update(
+        self,
+        features: Sequence[Dict],
+        accessor: Callable,
+        on_top: bool = True,
+        **kwargs
+    ):
+        """Update existing MosaicJSON on backend."""
+        version = self.mosaic_def.version
+        if version:
+            uversion = list(map(int, version.split(".")))
+            uversion[-1] += 1
+            version = ".".join(map(str, uversion))
+        else:
+            version = "1.0.0"
+
+        updated_quadkeys = set()
+
+        dataset_geoms = polygons(
+            [feat["geometry"]["coordinates"][0] for feat in features]
+        )
+
+        for idx, feature in enumerate(features):
+            tiles = burntiles.burn([feature], self.quadkey_zoom)
+            tiles = [mercantile.Tile(*tile) for tile in tiles]
+
+            tree = STRtree([dataset_geoms[idx]])
+
+            for tile in tiles:
+                quadkey = str(mercantile.quadkey(tile))
+                tile_geom = polygons(
+                    mercantile.feature(tile)["geometry"]["coordinates"][0]
+                )
+
+                # Find intersections from rtree
+                intersections_idx = sorted(
+                    tree.query(tile_geom, predicate="intersects")
+                )
+                if len(intersections_idx) == 0:
+                    continue
+
+                intersect_dataset, intersect_geoms = zip(
+                    *[(features[idx], dataset_geoms[idx]) for idx in intersections_idx]
+                )
+
+                dataset = self.mosaic_def._filter(
+                    tile, intersect_dataset, intersect_geoms, **kwargs
+                )
```

Comment:
for now on `minimum_tile_cover` is a valid kwargs I'll refactor a bit to first read the tile dataset ?

Reply:
oh I remember why I don't read the `assets` before, it's because we don't have the geometry of them, so we can not apply any other filtering (e.g `minimum_tile_cover`, `tile_cover_sort`) on the full list (old + new assets)

Reply:
Yeah I think `_filter` here is weird. Because in general `_filter` expects to take the entire dataset. I'm not sure whether it really applies here.

Also, since this is only applied on _new_ assets, if you pass `maximum_items_per_tile=5`, that restricts to 5 _new_ assets, so your mosaic could grow by 5 items per tile every time you call `update`.

Reply:
yeah basically this is why I only expose `minimum_tile_cover` because we cannot have the whole stack of dataset. This is totally weird but has to do with the new implementation